packages/flutter/test/material/data_table_test.dart

```diff
@@ -1691,6 +1691,53 @@ void main() {
     await gesture.up();
   });
 
+  testWidgetsWithLeakTracking('Material3 - DataRow renders custom colors when pressed', (WidgetTester tester) async {
+    const Color pressedColor = Color(0xff4caf50);
+    Widget buildTable() {
+      return DataTable(
+        columns: const <DataColumn>[
+          DataColumn(
+            label: Text('Column1'),
+          ),
+        ],
+        rows: <DataRow>[
+          DataRow(
+            color: MaterialStateProperty.resolveWith<Color>(
+              (Set<MaterialState> states) {
+                if (states.contains(MaterialState.pressed)) {
+                  return pressedColor;
+                }
+                return Colors.transparent;
+              },
+            ),
+            onSelectChanged: (bool? value) {},
+            cells: const <DataCell>[
+              DataCell(Text('Content1')),
+            ],
+          ),
+        ],
+      );
+    }
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(useMaterial3: true),
```

Comment:
nit: since the `useMaterial3` defaults to true now, we probably can just leave here as `ThemeData()` in the future:)